pkg/controller/virtualnode.go

```diff
@@ -211,6 +213,33 @@ func vnodeNeedsUpdate(desired *appmeshv1beta1.VirtualNode, target *aws.VirtualNo
 			return true
 		}
 	}
+
+	if desired.Spec.Logging != nil {
+		if target.Data.Spec.Logging == nil {
+			return true
+		}
+		if desired.Spec.Logging.AccessLog != nil {
+			if target.Data.Spec.Logging.AccessLog == nil {
+				return true
+			}
+			if desired.Spec.Logging.AccessLog.File != nil {
+				if target.Data.Spec.Logging.AccessLog.File == nil {
+					return true
+				}
+				if len(desired.Spec.Logging.AccessLog.File.Path) > 0 {
+					if target.Data.Spec.Logging.AccessLog.File.Path == nil {
+						return true
+					}
+					if desired.Spec.Logging.AccessLog.File.Path != awssdk.StringValue(target.Data.Spec.Logging.AccessLog.File.Path) {
+						return true
+					}
+				}
+			}
+		}
+	} else if target.Data.Spec.Logging != nil {
+		return true
+	}
```

Comment:
Lots of cases here, but they all add up. Could be worth adding comments, but not a must.

Reply:
separated out into its own method